### PR TITLE
Link Jobs to Transactions

### LINKED ISSUE #134: Add more detail to Jobs/CybersourceBankReconcile
Add a filter to display only jobs with transactions and a column in the data table for number of transactions.

### PR DESCRIPTION
**Pull request by sprucely**
Adds some foreign keys from Transactions to the JobRecord tables.
Adds transactions table to Job details pages.
Adds a checkbox filter limiting results to Jobs with transactions.

closes #134

### CODE REVIEW

**Comment by jSylvestre** [comment]
@sprucely can you make the filter display a little larger?
image

File: Sloth.Core/Services/CyberSourceBankReconcileService.cs [hunk 1]
```diff
@@ -151,13 +156,14 @@ private async Task ProcessReport(Report report, string reportXml, Integration in
...
-                            ProcessorTrackingNumber = deposit.RequestID,
-                            TransactionDate         = deposit.LocalizedRequestDate,
+                            Source                    = integration.Source,
+                            Status                    = TransactionStatuses.Scheduled,
+                            DocumentNumber            = documentNumber,
+                            KfsTrackingNumber         = kfsTrackingNumber,
+                            MerchantTrackingNumber    = deposit.MerchantReferenceNumber,
+                            ProcessorTrackingNumber   = deposit.RequestID,
+                            TransactionDate           = deposit.LocalizedRequestDate,
+                            CybersourceBankReconcileJobRecordId = jobRecord.Id
```
Comment (srkirkland) [inline Sloth.Core/Services/CyberSourceBankReconcileService.cs:166]: FWIW the formatting looks a little off here now, the assignments don't line up.  I think John had a plugin that would do that automatically.  I don't care as much about it, but it does look weird now that all of them are lined up except the last one.

**Review by srkirkland — APPROVED** [review]

File: Sloth.Core/Services/CyberSourceBankReconcileService.cs (hunk 1 above)
Comment (jSylvestre) [reply Sloth.Core/Services/CyberSourceBankReconcileService.cs:166]: I use Code Alignment I think.

Comment (sprucely) [reply Sloth.Core/Services/CyberSourceBankReconcileService.cs:166]: In this particular example, I had aligned it, but the property didn't contain the word "Reconcile". I later refactored it elsewhere, and boom.

**Pull request merged**